Requiring implementation experience for bindings (#414)
w3c/wot-binding-templates: registry-requirements.md

    - Implementing a non-commercial Consumer application/driver
    - Conditions for commercial use: E.g. building a commercial product with the binding
    - Making a statement about your product's supporting that binding
9. Transition from "Initial" to "Current"
  - Starting from the initial submission each binding has to demonstrate a certain level of concrete development maturity. This process involves real-world testing, which can take place in Plugfests, independent testing events, or even informal collaboration between developers. These testing events do not have to be organized by W3C and can be conducted remotely, including over VPN. The goal is to demonstrate that the binding correctly maps protocol operations and is well understood by at least two parties.

>   - At each testing event every operation defined in the binding MUST be validated automatically (e.g., scripts, test suites, etc.) and the results SHOULD be published in a dedicated document (README, or other human-readable documents) called *Test Report*.

Reviewer: > the results SHOULD be published in a dedicated document

There is a small logical gap. We do not mandate the existence of a test report anywhere. So an event doesn't need a test report output but at least one test report should exist in the end.

  - A *Test Report*
      - MUST contain information on the testing environment

>       - MUST provide an example of the logical process (not necessarily code) about how a TD can be processed to establish a communication between consumer and exposer.

Reviewer: This was intended to be actually more like a use case, e.g. controlling temperature in a room by sending requests. However, what you are requiring also makes sense

Reviewer: TD Call: this makes more sense as part of the binding itself as it does not change per implementation

Reviewer: ```suggestion
      - MUST provide an example of the logical process (not necessarily code) about how a TD can be processed to establish a communication between consumer and exposer.
      - MUST contain information about the scenario that was tested, e.g. controlling the room temperature by measuring temperature and adjusting the heater.
```

      - MUST explain where discussions on implementation experience should be collected

>       - SHOULD provide the history of all the past events (or explain how to retrieve the history of the results gathered during those events)

Reviewer: by event you mean testing event right? Not event affordance.

>       - SHOULD contain a reference to possible implementations of Cosumers or Exposers. 

Reviewer 2: ```suggestion
      - SHOULD contain a reference to possible implementations of Consumers or Exposers. 
```

>       - SHOULD contain a reference to possible implementations of Consumers or Exposers. 

Reviewer: ```suggestion
      - SHOULD contain a reference to the implementations of Consumers or Exposers. 
```

>   - For the binding to transition to the Current state, the test report MUST contain at least one implementation of a Consumer (capable of understanding and performing all the operations described in the binding) and one Exposer (capable of creating TDs with all the operations and features described in the binding). Additional implementations can be added even after the transition to the Current

Reviewer 2: ```suggestion
  - For the binding to transition to the "Current" state, the test report MUST contain at least one implementation of a Consumer (capable of understanding and performing all the operations described in the binding) and one Exposer (capable of creating TDs with all the operations and features described in the binding). Additional implementations can be added even after the transition to the Current
```

>   - For the binding to transition to the "Current" state, the test report MUST contain at least one implementation of a Consumer (capable of understanding and performing all the operations described in the binding) and one Exposer (capable of creating TDs with all the operations and features described in the binding). Additional implementations can be added even after the transition to the Current

Reviewer: I would not say that Exposer should be able to create TDs. TDs can come from another entity than the Thing

>   - For the binding to transition to the "Current" state, a *Test Report* MUST exists and MUST contain at least one implementation of a Consumer (capable of understanding and performing all the operations described in the binding) and one Exposer (capable of handling all the operations and features described in the binding and optionally be able to create a valid TD). Additional implementations can be added even after the transition to the Current

Reviewer: ```suggestion
  - For the binding to transition to the "Current" state, a *Test Report* MUST exist. The *Test Report* MUST contain at least one implementation of a Consumer (capable of understanding and performing all the operations described in the binding) and one Exposer (capable of handling all the operations and features described in the binding and optionally be able to create a valid TD). Additional implementations can be added even after the transition to the Current
```

>   - Submitters can call for transition but the custodian can also automatically trigger the process once it is verified that the condition above is reached.

Reviewer: ```suggestion
  - Submitters MAY call for transition but the custodian can also automatically trigger the process once it is verified that the condition above is reached.
```

>   - *Test Report*s and the related resources SHOULD be published in a git repository. The repository SHOLD be public and it MUST be accessible to the reviewers and the custodian.  

Reviewer 2: ```suggestion
  - *Test Report*s and the related resources SHOULD be published in a git repository. The repository SHOULD be public and it MUST be accessible to the reviewers and the custodian.  
```

      - SHOULD contain a reference to possible implementations of Consumers or Exposers. 
  - For the binding to transition to the "Current" state, the test report MUST contain at least one implementation of a Consumer (capable of understanding and performing all the operations described in the binding) and one Exposer (capable of creating TDs with all the operations and features described in the binding). Additional implementations can be added even after the transition to the Current
  - Submitters can call for transition but the custodian can also automatically trigger the process once it is verified that the condition above is reached.

>   - *Test Report*s and the related resources SHOULD be published in a git repository. The repository SHOULD be public and it MUST be accessible to the reviewers and the custodian.  

Reviewer 3: ```suggestion
  - *Test Reports* and related resources SHOULD be published in a git repository. The repository SHOULD be public and it MUST be accessible to the reviewers and the custodian.  
```
